service/src/main/java/skills/auth/saml/UserController.groovy

```diff
@@ -0,0 +1,47 @@
+/**
+ * Copyright 2020 SkillTree
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package skills.auth.saml
+
+import groovy.util.logging.Slf4j
+import jakarta.servlet.http.HttpServletRequest
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.context.annotation.Conditional
+import org.springframework.web.bind.annotation.*
+import skills.auth.SecurityMode
+import skills.auth.UserAuthService
+import skills.controller.exceptions.SkillsValidator
+
+@Conditional(SecurityMode.SAML2Auth)
+@RestController
+@RequestMapping("/")
+@Slf4j
+class UserController {
```

Comment:
instead of introducing `UserController.groovy` and duplicating the code from `CreateAccountController` please remove `@Conditional(SecurityMode.NonSAML2Auth)` from `CreateAccountController`, after that `UserController.groovy` can be simply removed

Reply:
Refactored as per your suggestions